sound/soc/intel/boards/skl_hda_dsp_generic.c

```diff
@@ -135,11 +138,20 @@ static int skl_hda_fill_card_info(struct snd_soc_acpi_mach_params *mach_params)
 				skl_hda_be_dai_links[IDISP_DAI_COUNT +
 					HDAC_DAI_COUNT + i];
 		}
-	} else if (codec_count == 2 && codec_mask & IDISP_CODEC_MASK) {
+	} else if (codec_count <= 2) {
 		num_links = ARRAY_SIZE(skl_hda_be_dai_links);
 		num_route = ARRAY_SIZE(skl_hda_map);
 		card->dapm_widgets = skl_hda_widgets;
 		card->num_dapm_widgets = ARRAY_SIZE(skl_hda_widgets);
+		if (!(codec_mask & IDISP_CODEC_MASK)) {
+			if (codec_count == 2)
+				return -EINVAL;
+			for (i = 0; i < IDISP_DAI_COUNT; i++) {
+				skl_hda_be_dai_links[i].codecs = dummy_codec;
+				skl_hda_be_dai_links[i].num_codecs =
+					ARRAY_SIZE(dummy_codec);
+			}
+		}
```

Comment:
agree with @lgirdwood we should probably add comments to explain the supported configurations, so that reviewers can just look at the code without having to reverse-engineer the logic.

I think we care about the three cases, yes?

a) HDMI only
b) HDAudio with no HDMI
c) HDAudio with HDMI.

Reply:
Ack @plbossart @lgirdwood . I'll make it clearer we handle these three specific cases.